src/event-stream/tokens-contract-handler.ts

```diff
@@ -626,27 +643,77 @@ class TokensContractHandler {
   /**
    * Make readonly contract call
    */
-  private async makeReadOnlyContractCall(
+  async makeReadOnlyContractCall(
+    functionName: string,
+    functionArgs: ClarityValue[]
+  ): Promise<ReadOnlyContractCallResponse> {
+    const url = this.stacksNetwork.getReadOnlyFunctionCallApiUrl(
+      this.contractAddress,
+      this.contractName,
+      functionName
+    );
+    const args = functionArgs.map(arg => cvToHex(arg));
+
+    const body = JSON.stringify({
+      sender: this.address,
+      arguments: args,
+    });
+
+    const response = await fetchPrivate(url, {
```

Comment:
My understanding is that both this `await` and the below `await response.json()` can throw if there's a network error, e.g. dns resolution failure, host unreachable, connection interrupted, etc.

The error handling below checking `response.ok` and `response.status` is only applicable the request was successful at the http protocol layer (i.e. headers were received without interrupt, then the body was received without interrupt).

Reply:
So I think that would require a try/catch block around these statements. You should be able to test these kind of network failures in a unit test. There should be a test for network unreachable (like using a bogus hostname), another test should be for connection interrupt i.e. have an http server reply with the headers, flush connection data, then force close the socket without sending the http response body.